java/vector/src/main/java/org/apache/arrow/vector/types/Types.java

```diff
@@ -639,6 +639,9 @@ public MinorType visit(FloatingPoint type) {
       }
 
       @Override public MinorType visit(Time type) {
+        if (type.getUnit() != TimeUnit.MILLISECOND || type.getBitWidth() != 32) {
+          throw new IllegalArgumentException("Only milliseconds on 32 bits supported for now: " + type);
+        }
```

Comment:
Will you need to do something like what was done with timestamps for multiple unit support?